src/menus/menus.service.ts

```diff
@@ -160,30 +160,67 @@ export class MenusService {
   }
 
   async updateOne(id: string, menuData: CreateMenuDto) {
-    await this.menuModel.updateOne({ _id: id }, menuData).exec();
+    const oldMenu = await this.menuModel.findByIdAndUpdate(id, menuData).exec();
+    if (oldMenu.category._id === menuData.category) {
+      return;
+    }
+
+    await this.categoriesService.pullMenuFromCategory(
+      oldMenu.category._id,
+      oldMenu._id,
+    );
+
+    if (!menuData.category) {
+      return;
+    }
+
+    await this.categoriesService.pushMenuToCategory(
+      menuData.category._id,
+      oldMenu._id,
+    );
   }
 
   async deleteOneMenu(id: string) {
     const currentDate = new Date();
-    await this.menuModel.updateOne({ _id: id }, { deleted_at: currentDate });
+    const menu = await this.menuModel
+      .findByIdAndUpdate(
+        id,
+        {
+          deleted_at: currentDate,
+        },
+        { new: true },
+      )
+      .exec();
+    if (menu.category) {
+      await this.categoriesService.pullMenuFromCategory(
+        menu.category._id,
```

Comment:
category isn't populated. It does not have _id in `menu.category`but `menu.category` is ObjectID simpily `<Types.ObjectId>menu.category`.